netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -79,14 +79,18 @@
    * noticing changes to DNS.
    */
   public static NettyChannelBuilder forAddress(SocketAddress serverAddress) {
```

Comment:
The caveat of using this method is already in the javadoc. I don't think there is anything to be fixed.
Making a special case for InetSocketAddress would obscure the semantics of this method, which I don't feel right.